app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -227,6 +227,7 @@ def handle_unsupported_id_type(enrollment, response)
       primary_id_type: response['primaryIdType'],
       reason: 'Unsupported ID type',
       job_name: self.class.name,
+      tmx_status: enrollment&.profile&.fraud_pending_reason || :threatmetrix_pass,
```

Comment:
**Nit**: Do you need to do the safe navigation operator on the enrollment since it's coming in as an argument?

Reply:
I need to check my Ruby syntax, I'm rusty. Here the `&.` safe operator is applying to `profile` and `fraud_pending_reason`. I wasn't sure if `profile` was always defined. I see in this job that elsewhere there's a usage without the safety net. In `enrollment_helper.rb` there isn't. 

Should I go ahead and remove all the safeties from profile? It was me being extra secure because I wasn't sure.

Reply:
The syntax can be a little counter-intuitive. `foo&.bar` is roughly equivalent to `defined?(foo) ? foo.bar : nil` — it checks for `foo` being nil, not `bar`, to prevent "Undefined method bar for foo" exceptions. I think we want that guard for `profile`, but don't need it on `enrollment`.

This is a weird case, because really it's nonsensical to have an enrollment with no profile, but it's possible to get into that state. Not all code does that check; we started adding it after someone managed to end up with a nil profile and it broke the job. (It actually just caused it to trigger an alert every hour, but it did process the other enrollments.)

So I think this could be `enrollment.profile&.fraud_pending_reason || :threatmetrix_pass,`. I think a null enrollment might be truly 'exceptional'. That said, I don't really feel strongly about a superfluous nil-safe check.

Reply:
Yeah, sorry for blowing this up. I think the thing for me is what do we want to do if the enrollment is nil? With the safe navigator check in place we get the symbol returned, but as @n1zyy said, maybe we want to trigger an exception? BUT THIS IS ME BIKESHEDDING. Please feel free to ignore me.